plugins/ceph_osd_plugin.py

```diff
@@ -87,23 +90,40 @@ def get_stats(self):
                 osd_data['in'] += 1
             else:
                 osd_data['out'] += 1
-    
+
         return data
 
+    def get_stats_via_tool(self):
+        """Retrieves stats using a subprocess."""
+        raw_output = subprocess.check_output(
+            ['ceph', 'osd', 'dump', '--format', 'json', '--cluster ', self.cluster])
+        return json.loads(raw_output)
+
+    @staticmethod
+    def get_stats_via_rest():
```

Comment:
should be generalized in base plugin. we want also metrics for pools / cluster

Reply:
Generalization in the base will be harder because we'll need to map the command each plugin invokes right now, with the appropriate rest command.
I think for now this is a good compromise